ngi_reports/stockholm/project_summary.py

```diff
@@ -161,64 +163,68 @@ def __init__(self, config, LOG, working_dir, **kwargs):
             clus_meth = ["cBot","onboard clustering"][seq_plat == "MiSeq" or fc_runp.get("ClusteringChoice","") == "OnBoardClustering"]
-            casava = fc_obj.get('DemultiplexConfig',{}).values()[0].get('Software',{}).get('Version')
+            try:
+                casava = fc_obj.get('DemultiplexConfig',{}).values()[0].get('Software',{}).get('Version')
+            except:
+                casava = None
             if seq_plat == "MiSeq":
                 seq_software = "MSC {}/RTA {}".format(fc_runp.get("MCSVersion"),fc_runp.get("RTAVersion"))
-                self.proj_has_miseq = True
             else:
                 seq_software = "{} {}/RTA {}".format(fc_runp.get("ApplicationName"),fc_runp.get("ApplicationVersion"),fc_runp.get("RTAVersion"))
-                self.proj_has_hiseq = True
             tmp_method = seq_template.format("SECTION", clus_meth, seq_plat, seq_software, run_setup, fc_chem, casava)
-            
+
             ## to make sure the sequencing methods are unique
-            if tmp_method not in seq_methods.keys():
-                seq_methods[tmp_method] = alphabets[len(seq_methods.keys())]
-            self.flowcell_info[fc_name]['seq_meth'] = seq_methods[tmp_method]
-            
+            if tmp_method not in self.seq_methods.keys():
+                self.seq_methods[tmp_method] = alphabets[len(self.seq_methods.keys())]
+            self.flowcell_info[fc_name]['seq_meth'] = self.seq_methods[tmp_method]
+
             ## Collect quality info for samples and collect lanes of interest
             for stat in fc_illumina.get('Demultiplex_Stats',{}).get('Barcode_lane_statistics',[]):
+#                import pdb; pdb.set_trace()
                 try:
-                    if stat['Project'].replace('__','.') != self.project_name:
+                    if re.sub('__|_','.',stat['Project'],1) != self.project_name and stat['Project'] != self.project_name:
                         continue
-                    sample, lane = (stat['Sample ID'], stat['Lane'])
-                    ## to put in a empty dict for the first time
-                    sample_qval[sample] = sample_qval.get(sample,{})
+                    sample, lane = (stat['Sample'] if seq_plat == "HiSeqX" else stat['Sample ID'], stat['Lane'])
                     try:
-                        sample_qval[sample]['{}_{}'.format(lane, fc_name)] = {'qval': float(stat.get('% of >= Q30 Bases (PF)')),
-                                                                        'bases': int(stat.get('# Reads').replace(',',''))*int(run_setup.split('x')[-1])}
-                    except (TypeError, ValueError) as e:
+                        self.sample_qval[sample]['{}_{}'.format(lane, fc_name)] = {'qval': float(stat.get(sample_qstat_keys[seq_plat]['qval'])),
+                                                                        'bases': int(stat.get(sample_qstat_keys[seq_plat]['bases']).replace(',',''))*int(run_setup.split('x')[-1])}
```

Comment:
Personal again, but I'd break this up for the sake of readability. Also remove the above code L`176`-`178`.
```python
## different FC db have different key names, this will be changed when merging the dbs for X and 2500 FC
if seq_plat in ['HiSeq2500','MiSeq']:
  qval = float(stat.get('% of >= Q30 Bases (PF)'))
  base = stat.get('bases':'# Reads')
elif seq_plat == 'HiSeqX':
  qval = float(stat.get('% >= Q30bases'))
  base = stat.get('PF Clusters')
r_idx = '{}_{}'.format(lane, fc_name)
qval = float(stat.get(sample_qstat_keys[seq_plat]['qval']))
base = int(base.replace(',','')) * int(run_setup.split('x')[-1])
self.sample_qval[sample][r_idx] = {'qval': qval, 'bases': base}
```

Took me quite a long time to figure out what was going on here. Not sure if my version is any easier to understand or not :laughing: